### PR TITLE
Test advanced search options

### PR DESCRIPTION
**Pull request by mddilley**
This PR:
- Adds the setup for tests using React Testing Library (libraries, provider mocks, etc.)
- Adds tests around the Project List View autocomplete with lookup table options using React Testing Library. 

 Happy to talk about how this works and answer any questions as best as I can.

Some resources that I used:
- https://testing-library.com/docs/react-testing-library/intro/
- https://typeofnan.dev/mocking-apollo-graphql-queries-in-react-testing/
- https://www.apollographql.com/docs/react/development-testing/testing/#the-mockedprovider-component
- https://kentcdodds.com/blog/avoid-nesting-when-youre-testing

## Testing
**URL to test:**
This can only be tested locally

**Steps to test:**
1. Run `npm test` in the Moped editor project folder

---
#### Ship list
- [x] Code reviewed 
- [ ] Product manager approved
- [ ] Added to [QA test script](https://docs.google.com/spreadsheets/d/1n_O6MLh9cwwPf57HUM394Ea-z9uuoEV1-QW4axNZXLE/edit#) if applicable

### CODE REVIEW

File: moped-editor/src/components/GridTable/GridTableSearch.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ import { useLazyQuery } from "@apollo/client";
 import { format } from "date-fns";
 import { get } from "lodash";
 
-const useStyles = makeStyles(theme => ({
+const useStyles = makeStyles((theme) => ({
```
Comment (mddilley) [inline moped-editor/src/components/GridTable/GridTableSearch.js:28]: All these updates in `GridTableSearch` are from Prettier 💅

File: moped-editor/package.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
     "build:production": "env-cmd -e production npm run build:local",
     "build:netlifypr": "env-cmd -e netlifypr npm run build:local",
     "build:netlifybranch": "./set_env_branch_build.py",
-    "test": "react-scripts test",
+    "test": "react-scripts test --verbose",
```
Comment (mddilley) [inline moped-editor/package.json:25]: This gives us more output about each individual test

File: moped-editor/src/components/GridTable/GridTableFilters.js [hunk 3]
```diff
@@ -520,6 +520,7 @@ const GridTableFilters = ({
                       handleFilterOperatorClick(filterId, e.target.value)
                     }
                     label="field"
+                    data-testid="operator-select"
```
Comment (mddilley) [inline moped-editor/src/components/GridTable/GridTableFilters.js:523]: Finding the right part of some of the MUI components to interact was tricky so these props gives us a way to find them using [ByTestId](https://testing-library.com/docs/queries/bytestid)

**Review by chiaberry — APPROVED** [review]
This is so awesome!! Such a big lift to mock all the providers and lay the groundwork for expanding our tests. 🏆

**Review by frankhereford — APPROVED** [review]
checkmark

So many checkmarks! ✅

This worked great; thank you for writing these, and even more, for establishing the patterns for everyone to use.

**Pull request merged**